### PR TITLE
Move CI to GitHub Actions

### PR DESCRIPTION
**Pull request by malfet**
This PR migrates CircleCI workflow to GHA and makes a few minor changes:
 - Upload intermediary docker containers from GCR to ECR (as they are faster to download and do not incur egress costs to the infra)
 - For the same reason, workflows no longer require AWS credentials, as runners provision to have write access to the bucket/ECR
 - Skip the logic of rebasing PRs against latest trunk, as GH is doing it automatically
 - Build/publish docs on trunk in parallel to CPU tests build
 - Rely on GitHub reusable workflows and split workflow into `_build.yml`, `_test.yml` and `_docs.yml`
 
Test plan:
 - Enable codecov collection on PRs and test that it works (see [this](https://github.com/pytorch/xla/actions/runs/5064520203/jobs/9092527785?pr=5045) log)
 - Test docs build (see [this](https://github.com/pytorch/xla/actions/runs/5071034648/jobs/9106921476) log)

TODO: 
 - [x] Test/migrate doc-push step
 - [x]  Test/migrate code coverage collection step
 - [x] Delete `config.yml` and no-longer used steps from `.circleci/common.sh`

### CODE REVIEW

**Comment by JackCaoG** [comment]
@malfet Test is green now, is this one ready to be reviewed?

**Comment by malfet** [comment]
@JackCaoG it's missing codecov and doc push steps, working on adding them now, and than it'll be ready

**Comment by malfet** [comment]
@JackCaoG I think it is ready now with a small caveat of adding a doc push step, which if you don't mind I'll do after current CI run finishes (want to make sure that codecov upload logic works as expected)

**Comment by JackCaoG** [comment]
Thanks @malfet this is super exciting. I will take a look tmr after test is green.

**Comment by malfet** [comment]
Hmm, interestingly job OOMs, (even though `linux.4xlarge` should have 32Gb of RAM), will try to upgrade to `linux.12xlarge`, which has 96Gb of RAM

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    needs: build
+    with:
+      docker-image: ${{ needs.build.outputs.docker-image }}
+      runner: linux.8xlarge.nvidia.gpu
+      collect-coverage: true
+    secrets:
+      gcloud-service-key: ${{ secrets.GCLOUD_SERVICE_KEY }}
+
+  push-docs:
+    name: "Build & ublish docs"
```
Comment (atalman) [inline .github/workflows/build_and_test.yml:68]:
```suggestion
    name: "Build & publish docs"
```

Comment (malfet) [reply .github/workflows/build_and_test.yml:68]: I know, but I'm waiting for a review and clear signal before pushing the fix for the typo

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Comment by Subbarker** [comment]
was perusing this file for an example to work off of and noticed this typo: `metadtaa.json`

**Comment by malfet** [comment]
@Subbarker thanks, fixing in https://github.com/pytorch/xla/pull/5172

**Pull request merged**